packages/@aws-cdk/pipelines/test/pipeline.test.ts

```diff
@@ -32,26 +33,30 @@ afterEach(() => {
   app.cleanup();
 });
 
-test('references stack template in subassembly', () => {
-  // WHEN
-  pipeline.addApplicationStage(new OneStackApp(app, 'App'));
-
-  // THEN
-  expect(pipelineStack).toHaveResourceLike('AWS::CodePipeline::Pipeline', {
-    Stages: arrayWith({
-      Name: 'App',
-      Actions: arrayWith(
-        objectLike({
-          Name: 'Stack.Prepare',
-          InputArtifacts: [objectLike({})],
-          Configuration: objectLike({
-            StackName: 'App-Stack',
-            TemplatePath: stringLike('*::assembly-App/*.template.json'),
+complianceSuite('references stack template in subassembly', (tests) => {
```

Comment:
This is just a teaser. Once we agreed on the interface and everything, we can move all tests over to the compliance strategy.